_posts/2025-02-01-real-time-data-replication-with-debezium-and-python.adoc

```diff
@@ -0,0 +1,342 @@
+---
+
+When it comes to replicating operational data for analytics, Change Data Capture (CDC) is the gold standard.
+It offers scalability, near real-time performance, and captures all data modifications, ensuring your analytical datasets are always up-to-date.
+Debezium is a leading tool in this space, connecting to a wide range of databases and exporting CDC events in various formats like JSON and Avro, making integration with diverse systems a breeze.
+
+While Debezium itself is a Java-based project, the data engineering world increasingly relies on Python.
+This blog post demonstrates how to leverage Debezium within a Python environment, using https://github.com/memiiso/pydbzengine[pydbzengine].
+We'll explore how to use these technologies to build a robust and scalable CDC solution.
+
++++<!-- more -->+++
+
+== Python-Powered CDC Pipeline
+
+=== Capturing Change Data with Debezium and Loading into DuckDB using DLT
+
+This blog post demonstrates how to use Debezium to capture change data from a PostgreSQL database and load it into a DuckDB database using the Data Load Tool (DLT). **All of this is achieved through a Python-based data processing pipeline using https://github.com/memiiso/pydbzengine[pydbzengine]**.
+
+We'll walk through the code, explaining the key components and how they work together.
+
+=== Understanding the Components
+
+1. https://debezium.io/[**Debezium:**] A powerful open-source platform for change data capture.
+It monitors database transaction logs and produces change event streams, indicating inserts, updates, and deletes.
+
+2. https://github.com/memiiso/pydbzengine[**pydbzengine:**] A Python library that provides a convenient way to interact with the Debezium embedded engine. [.underline]#It simplifies the process of configuring and running Debezium within a Python application.#
+
+3. https://dlthub.com/[**DLT:**] A versatile data loading tool that simplifies the process of extracting, and loading data into various destinations.
+In this example, we use DLT to load the change events from Debezium into DuckDB.
+
+4. https://duckdb.org/[**DuckDB:**] An embeddable analytical database that provides efficient data processing capabilities.
+It's an excellent choice for local development and testing.
+
+5. https://testcontainers.com/[**Testcontainers:**] A library for spinning up throwaway instances of services, such as databases, for testing.
+We use it to manage the PostgreSQL database during the example.
+
+=== Code Breakdown
+
+The provided code demonstrates a complete CDC pipeline in pyton, from capturing changes with Debezium to loading them into DuckDB with DLT.
+Let's break down the key parts:
+
+==== 1. Setting up the Environment
+
+Let's begin with setting up the necessary environment, including defining file paths, cleaning up previous runs, and defining a helper class `DbPostgresql` that uses Testcontainers to manage the PostgreSQL database, which acts as our operational data source.
+
+[%nowrap,python]
```

Comment:
I wonder if this would look slightly better as a source block using `[source,python]` ?
Same for the other code blocks.

Reply:
updated it to ,  `[source,python,options="nowrap"]`